### PR TITLE
Switch to Go native error wrapping

### PR DESCRIPTION
**Pull request by kolyshkin**
* switch to native error wrapping
* add `go.mod`
* update go versions in `travis.yml`

Compatibility notes:

1. `ErrSymlinkLoop` is removed in favor of wrapping `syscall.ELOOP` into
       `&os.PathError{}`; users should use e.g. `errors.Is(err, syscall.ELOOP)`
       to check for this particular error.
    
2. `IsNotExist` no longer use `pkg/errors.Cause` but native go error
       unwrapping; it could breaks a use case when it is used for some
       third-party errors wrapped using `pkg/errors.Wrap[f]`.
    
I was not able to find any users of either feature.

This is a shorter/simpler alternative to #4. The differences are:
 * no intermediate vendoring of `pkg/errors`;
 * slightly simpler `IsNotExist` implementation;
 * removing `ErrSymlinkLoop`, using `os.PathError` wrapping to return `ELOOP`.
 
 @thaJeztah @cyphar PTAL

### CODE REVIEW

File: join.go [hunk 1]
```diff
@@ -12,39 +12,26 @@ package securejoin
...
-
-	"github.com/pkg/errors"
 )
 
 // ErrSymlinkLoop is returned by SecureJoinVFS when too many symlinks have been
 // evaluated in attempting to securely join the two given paths.
-var ErrSymlinkLoop = errors.Wrap(syscall.ELOOP, "secure join")
+//
+// Deprecated: use errors.Is(err, syscall.ELOOP) instead.
+var ErrSymlinkLoop = syscall.ELOOP
```
Comment (kolyshkin) [inline join.go:26]: I was not able to find any external user of this, so maybe it's better to just remove this?

Comment (kolyshkin) [reply join.go]: Removed

Comment (cyphar) [reply join.go]: Yeah I don't think anybody used it, but I added it so we could detect it in principle. `errors.Is(err, unix.ELOOP)` works now and is more idiomatic anyway.

**Review by cyphar — APPROVED** [review]
LGTM. :+1: to dropping everything in one go -- thanks for pushing this originally @thaJeztah!

**Comment by thaJeztah** [comment]
Thanks!

**Pull request merged**